src/pages/calculator/party/index.tsx

```diff
@@ -25,7 +26,17 @@ interface SelectedCharacter {
 }
 
 export const PartyAdd: React.FC & ComponentWithLayout = () => {
-  const allCharacters = useLiveQuery(queryAllCharacters)
+  const travelerGender: TravelerGender = useAppSelector(selectTravelerGender)
+
+  let allCharacters = useLiveQuery(queryAllCharacters)
+  if (Array.isArray(allCharacters)) {
+    // need this guard because sometimes allCharacters is undefined
+    console.log(travelerGender)
+    allCharacters = allCharacters.filter((character) =>
```

Comment:
When we need to conditionally act upon a data dependency (e.g. "wait for a dependency to exist"), this is usually why `useEffect` is used to track the value of a variable (often setting a local boolean `useState` to true/false depending on the value of the variable), but in this case it is not necessary -- there are two ways we can solve this:

1. We can use the default null gender returned by `useAppSelector(selectTravelerGender)` to force as a fallback gender (i.e. coerce null into either male or female). But this may cause flickering for split seconds if for example:
  - User previously chose "female" as gender (and is saved to local storage for subsequent site visits)
  - Note that the default state of the store is always null before it rehydrates (restoring state from local storage in new site visits)
  - There is a "split second" state change from null -> "female" from the state restoration process by redux (the "rehydration" process)
  - If the fallback for null was to show "male" icon, it may visibly flicker instantaneously from Aether to Lumine
2. Since we are forcing the users to select either male or female gender when they first use the app anyways, we can simply not render anything until both `allCharacters` and `travelerGender` exists / is not null. This is done already in line 89: `{allCharacters && allCharacters.map( ...` -- what this line currently does is it only maps to the `AvatarIcon` only if `allCharacters` first evaluates to a truthy value. We can simply extend this to an `{allCharacters && travelerGender && allCharacters.map(...)` call.